engine.go

```diff
@@ -406,6 +404,12 @@ func (engine *Engine) UpdatePeers() error {
 	}
 
 	Logger.Info(fmt.Sprintf("Updating peers %s (zone: %s)", engine.SecurityGroupName, engine.ZoneID))
+
+	vms, err := client.List(vm)
+	if err != nil {
+		return err
+	}
+
 	// grab the right to alter the Peers
 	engine.peersMu.Lock()
 	defer engine.peersMu.Unlock()
```

Comment:
also, let the API call(s) finish before you attempt to (ll)lock the list of _peers_

Reply:
👍